src/tests/ftest/launch.py

```diff
@@ -3126,6 +3127,14 @@ def main():
         args.slurm_setup = True
         args.user_create = True
 
+        # Limit repeated testing in CI
+        if args.repeat > MAX_CI_REPETITIONS:
+            logger.warning(
+                "The requested number of times to repeat the test(s) exceeds the CI limitation. "
+                "The --repeat argument has been reduced from %s to %s.",
+                args.repeat, MAX_CI_REPETITIONS)
+            args.repeat = MAX_CI_REPETITIONS
```

Comment:
It would be more work, but a more intelligent limit might take into account test timeout. E.g. short tests can be repeated more times.